src/Providers/UserProviders.jsx

```diff
@@ -1,45 +1,99 @@
-import React, { Component, createContext } from "react";
-import { auth, generateUserDocument } from "../firebaseCredentials.js";
+import React, { useEffect, useState, createContext } from "react";
+import { auth } from "../firebaseCredentials.js";
 import { httpGet } from "../lib/dataAccess";
 
 const baseEndpoint = process.env.REACT_APP_ADMIN_BASE_ENDPOINT;
 
-export const UserContext = createContext({ user: null });
-class UserProvider extends Component {
-    state = {
+export const UserContext = createContext();
+
+const isTokenExpired = (user) => {
+    const MILLISECONDS_PER_DAY = 24 * 3600 * 1000;
+    const NOW = new Date().getTime();
+
+    // Force user to relog in every 24 hours
+    return Number(user.toJSON().lastLoginAt) < NOW - MILLISECONDS_PER_DAY;
+};
+
+const AUTH_STATES = {
+    LOADING: "LOADING",
+    NOT_AUTHENTICATED: "NOT_AUTHENTICATED",
+    AUTHENTICATED: "AUTHENTICATED",
+};
+
+export default function UserProvider({ children }) {
+    const [authState, setAuthState] = useState({
         user: null,
-        isAdmin: false,
-    };
+        state: AUTH_STATES.LOADING,
+    });
 
-    componentDidMount = async () => {
-        auth.onAuthStateChanged(async (userAuth) => {
-            if (!userAuth) {
-                this.setState({ user: null });
+    useEffect(() => {
+        async function setUser(user) {
+            //Set the state to loading by default
+            setAuthState({ state: AUTH_STATES.LOADING, user: null });
+
+            if (user == null) {
+                setAuthState({ state: AUTH_STATES.NOT_AUTHENTICATED, user });
+                return;
             }
-            if (userAuth && userAuth.emailVerified) {
-                const user = await generateUserDocument(userAuth);
-                this.setState({ user });
+
+            if (isTokenExpired(user)) {
+                auth.signOut();
+                setAuthState({
+                    state: AUTH_STATES.NOT_AUTHENTICATED,
+                    user: null,
+                });
+                return;
             }
 
-            // validate cookies are set appropriately on this user instance
-            try {
-                const response = await httpGet(baseEndpoint + "user_profile");
-                this.setState({ isAdmin: response.status === 200 });
-            } catch {
-                this.setState({ isAdmin: false });
+            const response = await httpGet(baseEndpoint + "user_profile");
+
+            if (!response) {
+                setAuthState({
+                    state: AUTH_STATES.NOT_AUTHENTICATED,
+                    user: null,
+                });
+                return;
             }
+
+            setAuthState({
+                state: AUTH_STATES.AUTHENTICATED,
+                user: response.data,
+            });
+        }
+
+        const unsubscribe = auth.onAuthStateChanged(setUser);
```

Comment:
`unsubscribe`? What is this line's purpose?

Reply:
Probably not the best name, but essentially it sets up a listener so when auth.onAuthStateChanged happens, the `setUser` function is called (which is defined above on line 30).